File: code/backend-ms1/src/main/java/ma/zs/univ/service/facade/admin/commun/QuartierAdminService.java
package ma.zs.univ.service.facade.admin.commun;

import java.util.List;
import ma.zs.univ.bean.core.commun.Quartier;
import ma.zs.univ.dao.criteria.core.commun.QuartierCriteria;
import ma.zs.univ.zynerator.service.IService;



public interface QuartierAdminService extends  IService<Quartier,QuartierCriteria>  {

    List<Quartier> findBySecteurId(Long id);
    int deleteBySecteurId(Long id);
    long countBySecteurCode(String code);




}

File: code/frontend/src/app/shared/service/admin/commun/QuartierAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {QuartierDto} from 'src/app/shared/model/commun/Quartier.model';
import {QuartierCriteria} from 'src/app/shared/criteria/commun/QuartierCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class QuartierAdminService extends AbstractService<QuartierDto, QuartierCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): QuartierDto {
        return new QuartierDto();
    }

    public constrcutCriteria(): QuartierCriteria {
        return new QuartierCriteria();
    }

    get API() {
        return environment.apiUrlUnivservice + 'admin/quartier/';
    }
}
